File: scripts/generate_index.js
const fs = require("fs");
const path = require("path");

function directoryStructure(dir) {
    let structure = {};
    fs.readdirSync(dir).forEach(entry => {
        let fullPath = path.join(dir, entry);
        if (fs.lstatSync(fullPath).isDirectory()) {
            structure[entry] = directoryStructure(fullPath);
        } else {
            structure[entry] = entry;
        }
    });
    return structure;
}

const dataPaths = ["moments", "signals", "trendlines"];

for (const dataPath of dataPaths) {
    const directory = path.join(process.cwd(), dataPath);
    const structure = directoryStructure(directory);
    fs.writeFileSync(path.join(process.cwd(), "build", `${dataPath}.json`), JSON.stringify(structure, null, 4));
}

File: scripts/generate_index.py
import os
import json

def directory_structure(dir):
    structure = {}
    for entry in os.listdir(dir):
        path = os.path.join(dir, entry)
        if os.path.isdir(path):
            structure[entry] = directory_structure(path)
        else:
            structure[entry] = entry
    return structure

moments_dir = os.path.join(os.getcwd(), 'moments')
moments_structure = directory_structure(moments_dir)

with open(os.path.join(os.getcwd(), '_site', 'moments_index.json'), 'w') as file:
    json.dump(moments_structure, file, indent=4)